src/Hl7.Fhir.Support.Poco.Tests/NewPocoSerializers/FhirJsonDeserializationTests.cs

```diff
@@ -1049,6 +1050,144 @@ public void TestBackboneElementEmptyStack()
             var result = JsonSerializer.Deserialize<Parameters.ParameterComponent>(ref newJsonReader, options);
         }
+
+        private static IEnumerable<object[]> getExtensionOptionsAndExpectedErrors()
+        {
+            yield return
+            [
+                new JsonSerializerOptions().ForFhir(typeof(TestPatient).Assembly)
+                    .UsingMode(DeserializerModes.Ostrich),
+                new Predicate<IEnumerable<CodedException>>(errs => !errs.Any())
+            ];
+            yield return
+            [
+                new JsonSerializerOptions().ForFhir(typeof(TestPatient).Assembly)
+                    .UsingMode(DeserializerModes.Recoverable),
+                new Predicate<IEnumerable<CodedException>>(errs => !errs.Any(e => FilterPredicateExtensions.IsRecoverableIssue(e)))
+            ];
+            yield return
+            [
+                new JsonSerializerOptions().ForFhir(typeof(TestPatient).Assembly)
+                    .UsingMode(DeserializerModes.BackwardsCompatible),
+                new Predicate<IEnumerable<CodedException>>(errs => !errs.Any(e => FilterPredicateExtensions.IsBackwardsCompatibilityIssue(e)))
+            ];
+            yield return
+            [
+                new JsonSerializerOptions().ForFhir(typeof(TestPatient).Assembly)
+                    .Ignoring([ERR.UNEXPECTED_JSON_TOKEN_CODE]),
+                new Predicate<IEnumerable<CodedException>>(errs => errs.All(e => e.ErrorCode != ERR.UNEXPECTED_JSON_TOKEN_CODE))
+            ];
+            yield return
+            [
+                new JsonSerializerOptions().ForFhir(typeof(TestPatient).Assembly).UsingMode(DeserializerModes.Ostrich)
+                    .Enforcing([ERR.ARRAYS_CANNOT_BE_EMPTY_CODE, ERR.LONG_CANNOT_BE_PARSED_CODE]),
+                new Predicate<IEnumerable<CodedException>>(errs =>
+                {
+                    IEnumerable<CodedException> codedExceptions = errs as CodedException[] ?? errs.ToArray();
+                    return codedExceptions.Any() && codedExceptions.All(e =>
+                        e.ErrorCode is ERR.ARRAYS_CANNOT_BE_EMPTY_CODE or ERR.LONG_CANNOT_BE_PARSED_CODE);
+                })
+            ];
+        }
+        
+        [DataTestMethod]
+        [DynamicData(nameof(getExtensionOptionsAndExpectedErrors), DynamicDataSourceType.Method)]
+        public void TestExtensionMethods(JsonSerializerOptions options, Predicate<IEnumerable<CodedException>> shouldHold)
+        {
+            string testJson = File.ReadAllText(Path.Combine("TestData", "fp-test-patient-errors.json"));
+            
+            try
+            {
+                _ = JsonSerializer.Deserialize<TestPatient>(testJson, options);
+                throw new DeserializationFailedException(null, []);
+            }
+            catch (DeserializationFailedException dfe)
+            {
+                Assert.IsTrue(shouldHold(dfe.Exceptions));
```

Comment:
We should be using FluentAssertions for all new unit tests, so  `shouldHold(dfe.Exceptions)`.Should().BeTrue();`